it/src/main/java/org/corfudb/universe/node/server/docker/DockerCorfuServer.java

```diff
@@ -0,0 +1,403 @@
+        super(params, universeParams);
+        this.docker = docker;
+        this.loggingParams = loggingParams;
+    }
+
+    /**
+     * Deploys a Corfu server / docker container
+     */
+    @Override
+    public DockerCorfuServer deploy() {
+        log.info("Deploying the Corfu server. Docker container: {}", params.getName());
+
+        deployContainer();
+
+        return this;
+    }
+
+    /**
+     * This method attempts to gracefully stop the Corfu server. After timeout, it will kill the Corfu server.
+     *
+     * @param timeout a duration after which the stop will kill the server
+     * @throws NodeException this exception will be thrown if the server cannot be stopped.
+     */
+    @Override
+    public void stop(Duration timeout) {
+        log.info("Stopping the Corfu server. Docker container: {}", params.getName());
+
+        try {
+            ContainerInfo container = docker.inspectContainer(params.getName());
+            if (!container.state().running() && !container.state().paused()) {
+                log.warn("The container `{}` is already stopped", container.name());
+                return;
+            }
+            docker.stopContainer(params.getName(), (int) timeout.getSeconds());
+        } catch (DockerException | InterruptedException e) {
+            throw new NodeException("Can't stop Corfu server: " + params.getName(), e);
+        }
+    }
+
+    /**
+     * Immediately kill the Corfu server.
+     *
+     * @throws NodeException this exception will be thrown if the server can not be killed.
+     */
+    @Override
+    public void kill() {
+        log.info("Killing the Corfu server. Docker container: {}", params.getName());
+
+        try {
+            ContainerInfo container = docker.inspectContainer(params.getName());
+
+            if (!container.state().running() && !container.state().paused()) {
+                log.warn("The container `{}` is not running", container.name());
+                return;
+            }
+            docker.killContainer(params.getName());
+        } catch (DockerException | InterruptedException ex) {
+            throw new NodeException("Can't kill Corfu server: " + params.getName(), ex);
+        }
+    }
+
+    /**
+     * Immediately kill and remove the docker container
+     *
+     * @throws NodeException this exception will be thrown if the server can not be killed.
+     */
+    @Override
+    public void destroy() {
+        log.info("Destroying the Corfu server. Docker container: {}", params.getName());
+
+        if (destroyed.getAndSet(true)) {
+            log.debug("Already destroyed: {}", params.getName());
+            return;
+        }
+
+        try {
+            kill();
+        } catch (NodeException ex) {
+            log.warn("Can't kill container: {}", params.getName());
+        }
+
+        collectLogs();
+
+        try {
+            docker.removeContainer(params.getName());
+        } catch (DockerException | InterruptedException ex) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)